src/vs/workbench/contrib/preferences/browser/settingsTree.ts

```diff
@@ -2485,3 +2496,40 @@ class SyncSettingAction extends Action {
 	}
 
 }
+
+class ApplySettingToAllProfilesAction extends Action {
+	static readonly ID = 'settings.applyToAllProfiles';
+	static readonly LABEL = localize('applyToAllProfiles', "Apply Setting to all Profiles");
+
+	constructor(
+		private readonly setting: ISetting,
+		@IWorkbenchConfigurationService private readonly configService: IWorkbenchConfigurationService,
+	) {
+		super(ApplySettingToAllProfilesAction.ID, ApplySettingToAllProfilesAction.LABEL);
+		this._register(Event.filter(configService.onDidChangeConfiguration, e => e.affectsConfiguration(APPLY_ALL_PROFILES_SETTING))(() => this.update()));
+		this.update();
+	}
+
+	update() {
+		const allProfilesSettings = this.configService.getValue<string[]>(APPLY_ALL_PROFILES_SETTING);
+		this.checked = allProfilesSettings.includes(this.setting.key);
+	}
+
+	override async run(): Promise<void> {
+		// first remove the current setting completely from ignored settings
+		const value = this.configService.getValue<string[]>(APPLY_ALL_PROFILES_SETTING) ?? [];
+
+		if (this.checked) {
+			value.splice(value.indexOf(this.setting.key), 1);
```

Comment:
Is it possible for the `this.checked` value to become stale in between `update()` and `run()`?

Reply:
It is and can be ignored as this stale duration is negligible.